roles/client/tasks/main.yml

```diff
@@ -22,15 +22,43 @@
   retries: 5
   when: ursula_os == 'rhel' and openstack_install_method == 'distro'
 
+- block:
+  - name: make sure /opt/stack/ exists
+    file: dest=/opt/stack state=directory
+
+  - name: write constraints file
+    copy:
+      src: ../../openstack-source/files/constraints.txt
+      dest: /opt/stack/constraints.txt
+      force: true
+    when: client.upper_constraints == None
```

Comment:
As client.upper_constrain is defined in client/defaults/main.yml, so I think we don't need 'is defined' check.